### PR TITLE
update deps and prepare release v0.14.0

### PR DESCRIPTION
**Pull request by thyton**
update deps, bump .go-version to 1.22.6 to match vault repo and prepare release v0.14.0
```
go list -u -m -json all | jq -r 'select(.Indirect != true and .Update != null) | .Path+"@"+.Update.Version' | xargs -L1 go get
go get go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp@latest
go get github.com/hashicorp/go-secure-stdlib/plugincontainer@latest
go get github.com/hashicorp/go-retryablehttp@latest
go get github.com/ory/dockertest/v3@v3.10.0 \
  github.com/hashicorp/go-secure-stdlib/plugincontainer@v0.3.0 \
  github.com/docker/docker@v25.0.6 \
  github.com/docker/cli@v25.0.6
go mod tidy
```

### CODE REVIEW

**Comment by thyton** [comment]
Tests passed locally. CI failure is similar to https://github.com/hashicorp/vault-plugin-database-couchbase/actions/runs/10708495222/job/29690926954

```
$ make test
==> Checking that code complies with gofumpt requirements...
CGO_ENABLED=0 go test ./...  -timeout=20m
?   	github.com/hashicorp/vault-plugin-secrets-openldap/client/tools	[no test files]
?   	github.com/hashicorp/vault-plugin-secrets-openldap/cmd/vault-plugin-secrets-openldap	[no test files]
?   	github.com/hashicorp/vault-plugin-secrets-openldap/ldapifc	[no test files]
ok  	github.com/hashicorp/vault-plugin-secrets-openldap	(cached)
ok  	github.com/hashicorp/vault-plugin-secrets-openldap/client	(cached)
```

**Comment by tvoran** [comment]
I feel like we shouldn't ignore the CI failure, since that will make it hard to release?

Suggest downgrading to a docker/docker version that's compatible with the current vault/sdk tag, something like:

```
go get github.com/ory/dockertest/v3@v3.10.0 \
  github.com/hashicorp/go-secure-stdlib/plugincontainer@v0.3.0 \
  github.com/docker/docker@v25.0.6 \
  github.com/docker/cli@v25.0.6
go mod tidy
```

**Comment by thyton** [comment]
Thank you, @tvoran!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+
+* update dependencies [GH-113](https://github.com/hashicorp/vault-plugin-secrets-openldap/pull/113)
+  * `github.com/go-ldap/ldap/v3` v3.4.6 -> v3.4.8
+  * `github.com/hashicorp/go-hclog` v1.6.2 -> v1.6.3
+  * `github.com/hashicorp/go-secure-stdlib/parseutil` v0.1.7 -> v0.1.8
+  * `github.com/hashicorp/vault/api` v1.13.0 -> v1.14.0
+  * `github.com/hashicorp/vault/sdk` v0.12.0 -> v0.13.0
+  * `github.com/ory/dockertest/v3` v3.10.0 -> v3.11.0
+  * `golang.org/x/text` v0.14.0 -> v0.18.0
+  * `github.com/hashicorp/go-secure-stdlib/plugincontainer` v0.3.0 -> v0.4.0
```
Comment (tvoran) [inline CHANGELOG.md:15]: Probably should update these with the downgrade.

```suggestion
  * `golang.org/x/text` v0.14.0 -> v0.18.0
```

**Review by tvoran — APPROVED** [review]
May want to update `.go-version` as well, and I noticed there's a dependabot PR that probably should go in too: #111

File: CHANGELOG.md (hunk 1 above)
Comment (thyton) [reply CHANGELOG.md]: Thank you!

**Pull request merged**